### PR TITLE
[Encoder][Node] Add nil checks

### PR DESCRIPTION
**Pull request by siddimore**
## Why are these changes needed?

Add nil checks for incoming requests in Disperser and Node

Additional Protobuf checks in this PR: https://github.com/Layr-Labs/eigenda/pull/427

## Checks

- [ ] I've made sure the lint is passing in this PR.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, in that case, please comment that they are not relevant.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Integration tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by ian-shim — COMMENTED** [review]
can any of these proto requests themselves be nil?

File: node/node.go [hunk 1]
```diff
@@ -268,10 +268,15 @@ func (n *Node) ProcessBatch(ctx context.Context, header *core.BatchHeader, blobs
 	log.Debug("Processing batch", "num of blobs", len(blobs))
 
 	if len(blobs) == 0 {
-		return nil, errors.New("the number of blobs must be greater than zero")
+		return nil, errors.New("ProcessBatch: number of blobs must be greater than zero")
 	}
+
+	if len(rawBlobs) == 0 {
```
Comment (mooselumph) [inline node/node.go:274]: This check is unneeded; if `len(blobs) != 0` and `len(rawBlobs) == 0`, then the next check will fail.

**Review by mooselumph — APPROVED** [review]
lg! just one comment

File: node/grpc/utils.go [hunk 2]
```diff
@@ -67,6 +67,12 @@ func GetBlobMessages(in *pb.StoreChunksRequest) ([]*core.BlobMessage, error) {
 
 // GetBlobHeaderFromProto constructs a core.BlobHeader from a proto of pb.BlobHeader.
 func GetBlobHeaderFromProto(h *pb.BlobHeader) (*core.BlobHeader, error) {
+
+	if h == nil {
+		return nil, api.NewInvalidArgError("GetBlobHeaderFromProto: blob header is nil")
```
Comment (ian-shim) [inline node/grpc/utils.go:72]: Is this only used to handle API requests at the top level? 
Otherwise, i don't think it should return the api error

**Pull request merged**